src/openrct2/world/Location.hpp

```diff
@@ -702,6 +702,44 @@ struct TileCoordsXYZD : public TileCoordsXYZ
     }
 };
 
+struct BoundBoxXY
+{
+    CoordsXY offset{};
+    CoordsXY length{};
+
+    constexpr BoundBoxXY() = default;
+    constexpr BoundBoxXY(CoordsXY _offset, CoordsXY _length)
+        : offset(_offset)
+        , length(_length)
+    {
+    }
+
+    constexpr BoundBoxXY(int32_t _offX, int32_t _offY, int32_t _lenX, int32_t _lenY)
```

Comment:
I’d prefer if we dropped this constructor (and its counterpart for BoundBoxXYZ). Code should just use proper CoordsXY(Z) values.